plugins/CoreHome/javascripts/corehome.js

```diff
@@ -6,76 +6,73 @@
  */
 
 (function ($) {
-
+    //
+    // 'check for updates' behavior
+    //
     $(function () {
-
-        //
-        // 'check for updates' behavior
-        //
-
-        var headerMessageParent = $('#header_message').parent();
-
-        initTopControls();
-
-        // when 'check for updates...' link is clicked, force a check & display the result
-        headerMessageParent.on('click', '#updateCheckLinkContainer', function (e) {
-            var headerMessage = $(this).closest('#header_message');
-
-            var $titleElement = headerMessage.find('.title');
-            if ($titleElement.attr('target')) { // if this is an external link, internet access is not available on the server
-                return;
-            }
-
-            e.preventDefault();
-
-            var ajaxRequest = new ajaxHelper();
-            ajaxRequest.setLoadingElement('#header_message .loadingPiwik');
-            ajaxRequest.addParams({
-                module: 'CoreHome',
-                action: 'checkForUpdates'
-            }, 'get');
-
-            ajaxRequest.withTokenInUrl();
-
-            $titleElement.addClass('activityIndicator');
-
-            ajaxRequest.setCallback(function (response) {
-                headerMessage.fadeOut('slow', function () {
-                    response = $('#header_message', $('<div>' + response + '</div>'));
-
-                    $titleElement.removeClass('activityIndicator');
-
-                    var newVersionAvailable = response.hasClass('update_available');
-                    if (newVersionAvailable) {
-                        headerMessage.replaceWith(response);
-                        headerMessage.show();
-                    }
-                    else {
-                        headerMessage.find('.title').html(_pk_translate('CoreHome_YouAreUsingTheLatestVersion'));
-                        headerMessage.show();
-                        setTimeout(function () {
-                            headerMessage.fadeOut('slow', function () {
-                                headerMessage.replaceWith(response);
-                            });
-                        }, 4000);
-                    }
+        var initUpdateCheck = function() {
```

Comment:
Also since none of this code is part of the tracker JS where we do care about using legacy JS techniques, could we use `const` and `let` here?

Reply:
We could definitely.

I only skipped that to keep the style of the file sort of as is. And I think the next step should be to integrate it fully into the vue component, there should be no reason to keep this block around longer than necessary.